src/cls/cephfs/cls_cephfs.cc

```diff
@@ -0,0 +1,289 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2015 Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#include <string>
+#include <errno.h>
+#include <sstream>
+
+#include "objclass/objclass.h"
+
+CLS_VER(1,0)
+CLS_NAME(cephfs_size_scan)
+
+cls_handle_t h_class;
+cls_method_handle_t h_set_if_greater;
+cls_method_handle_t h_accumulate_inode_metadata;
+
+/**
+ * Set a named xattr to a given integer, if and only if the xattr
+ * is not already set to a greater integer.
+ *
+ * If the xattr is missing, or does not encode an integer, then
+ * it is set to the input integer.
+ *
+ * On success, the output buffer is populated with the resulting
+ * integer contained in the xattr.  On failure, a nonzero value is
+ * returned and the contents of the output buffer are undefined.
```

Comment:
Hm, are you actually getting this buffer back and populated in testing? Generally speaking we can't get any data back on write operations except for a return value.